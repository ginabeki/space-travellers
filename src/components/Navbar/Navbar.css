.navbar {
display: flex;
justify-content: space-around;
align-items: center;
align-content: center;
border-bottom: 1px solid #8F857D;
}

.logo {
    display: flex;
    justify-content: center;
    align-items: center;
}
.logoLink {
    text-decoration: none;
    color: #000;
    font-weight: bold;
    font-size: 1.5rem;
}

.menu-items {
    display: flex;
    list-style: none;
    gap: 1.5rem;
}

.navLink {
    text-decoration: none;
    color: #60D394;
    font-size: 1rem;
}