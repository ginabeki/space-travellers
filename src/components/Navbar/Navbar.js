import React from 'react';
import { Link } from 'react-router-dom';
import logo from '../Assets/space-logo.png';
import './Navbar.css';

const Navbar = () => (
  <header>
    <nav className="navbar">
      <div className="logo">
        <img src={logo} alt="logo" />
        <Link to="/" className="logoLink">Space Traveller&apos;s Hub</Link>
      </div>
      <ul className="menu-items">
        <li>
          <Link to="/rockets" className="navLink">Rockets</Link>
        </li>
        <li>
          <Link to="/mission" className="navLink">Missions</Link>
        </li>
        <li>
          <Link to="/profile" className="navLink">My Profile</Link>
        </li>

      </ul>
    </nav>
  </header>
);

export default Navbar;
