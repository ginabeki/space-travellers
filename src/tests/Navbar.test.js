import TestRenderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../components/Navbar/Navbar';

describe('Navbar Component Test', () => {
  it('Expected to render Navbar contenents', () => {
    const NavbarComponent = TestRenderer.create(
      <Router>
        <Navbar />
      </Router>,
    ).toJSON();
    expect(NavbarComponent).toMatchSnapshot();
  });
});
