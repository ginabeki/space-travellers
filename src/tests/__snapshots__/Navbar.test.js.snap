// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component Test Expected to render Navbar contenents 1`] = `
<header>
  <nav
    className="navbar"
  >
    <div
      className="logo"
    >
      <img
        alt="logo"
        src="space-logo.png"
      />
      <h2
        className="logoLink"
      >
        Space Traveller's Hub
      </h2>
    </div>
    <ul
      className="menu-items"
    >
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className=""
          href="/mission"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className=""
          href="/profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
